src/raftstore/store/worker/apply.rs

```diff
@@ -1096,14 +1190,16 @@ impl ApplyDelegate {
     }
 
     fn exec_split(
-        &mut self,
-        ctx: &mut ApplyContext,
+        self,
+        mut ctx: ApplyContext,
         req: &AdminRequest,
-    ) -> Result<(AdminResponse, Option<ExecResult>)> {
+    ) -> Box<Future<Item = AdminCmdFutureItem, Error = FutureError> + Send> {
```

Comment:
similar to the above case, seems that you can unify your return value type to the same type `future::result`, so the box can be dropped.